parlai/mturk/core/mturk_data_handler.py

```diff
@@ -0,0 +1,474 @@
+            'task_end': pairing_result[4],
+            'conversation_id': pairing_result[5],
+            'bonus_amount': pairing_result[6],
+            'bonus_text': pairing_result[7],
+            'bonus_paid': pairing_result[8],
+            'worker_id': pairing_result[9],
+            'assignment_id': pairing_result[10],
+            'run_id': pairing_result[11],
+        }
+
+    def _format_assignment(self, assign_result):
+        return {
+            'assignment_id': assign_result[0],
+            'status': assign_result[1],
+            'approve_time': assign_result[2],
+            'worker_id': assign_result[3],
+            'hit_id': assign_result[4],
+        }
+
+    def create_default_tables(self):
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute(CREATE_RUN_DATA_SQL_TABLE)
+        c.execute(CREATE_WORKER_DATA_SQL_TABLE)
+        c.execute(CREATE_HIT_DATA_SQL_TABLE)
+        c.execute(CREATE_ASSIGN_DATA_SQL_TABLE)
+        c.execute(CREATE_PAIRING_DATA_SQL_TABLE)
+        conn.commit()
+
+    def log_new_run(self, target_hits, task_group_id=None):
+        if task_group_id is None:
+            task_group_id = self.task_group_id
+        assert task_group_id is not None, 'Default task_group_id not set'
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('INSERT INTO runs VALUES (?,?,?,?,?);',
+                  (task_group_id, 0, target_hits, 0, 0))
+        conn.commit()
+
+    def log_hit_status(self, mturk_hit_creation_response, task_group_id=None):
+        '''Create or update an entry in the hit sattus table'''
+        if task_group_id is None:
+            task_group_id = self.task_group_id
+        assert task_group_id is not None, 'Default task_group_id not set'
+
+        hit_details = mturk_hit_creation_response['HIT']
+        id = hit_details['HITId']
+        expiration = time.mktime(hit_details['Expiration'].timetuple())
+        status = hit_details['HITStatus']
+        assignments_pending = hit_details['NumberOfAssignmentsPending']
+        assignments_available = hit_details['NumberOfAssignmentsAvailable']
+        assignments_complete = hit_details['NumberOfAssignmentsCompleted']
+        conn = self._get_connection()
+        c = conn.cursor()
+        c.execute('SELECT COUNT(*) FROM hits WHERE id = ?;', (id, ))
+        is_new_hit = c.fetchone()[0] == 0
+        if is_new_hit:
+            c.execute('UPDATE runs SET created = created + 1 WHERE id = ?;',
+                      (task_group_id, ))
+
+        c.execute('REPLACE INTO hits VALUES (?,?,?,?,?,?,?);',
+                  (id, expiration, status, assignments_pending,
+                   assignments_available, assignments_complete, task_group_id))
+        conn.commit()
+
+    def log_worker_accept_assignment(self, worker_id, assignment_id, hit_id,
+                                     task_group_id=None):
+        if task_group_id is None:
```

Comment:
since this is repeated a bunch, might just wanna make it a method e.g. `task_group_id = self.get_task_group_id(task_group_id)` and do the assert in there